File: FizzBuzz/Python_4/FizzBuzzTest.py
import unittest
import FizzBuzz


class FizzBuzzTest(unittest.TestCase):
    def setUp(self):
        self.sut = FizzBuzz.FizzBuzz()

    def test_Say_NumberNotDivisible3Or5_ReturnNumber(self):
        # arrange
        expected = '1'

        # act
        actual = self.sut.Say(1)

        # assert
        self.assertEqual(expected, actual)

    def test_Say_NumberDivisible3_ReturnFizz(self):
        # arrange
        expected = 'Fizz'

        # act
        actual = self.sut.Say(3)

        # assert
        self.assertEqual(expected, actual)
        
    def test_Say_NumberDisivible5_ReturnBuzz(self):
        # arrange
        expected = 'Buzz'

        # act
        actual = self.sut.Say(5)

        # assert
        self.assertEqual(expected, actual)

    def test_Say_NumberDisivibl3Ande5_ReturnFizzBuzz(self):
        # arrange
        expected = 'FizzBuzz'

        # act
        actual = self.sut.Say(15)

        # assert
        self.assertEqual(expected, actual)


if __name__ == '__main__':
    unittest.main

File: FizzBuzz/C#_6/FizzBuzzTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NUnit.Framework;

namespace Day6
{
    [TestFixture]
    class FizzBuzzTest
    {
        FizzBuzz _sut;

        [OneTimeSetUp]
        public void Init()
        {
            this._sut = new FizzBuzz();
        }

        [Test]
        public void Say_NumberNotDivisible_ReturnNumber()
        {
            // Arrange
            var number = 1;
            var expected = "1";

            // Act
            var actual = this._sut.Say(number);

            // Assert
            Assert.AreEqual(expected, actual);
        }

        [Test]
        public void Say_Divisible3_ReturnFizz()
        {
            // Arrange
            var number = 3;
            var expected = "Fizz";

            // Act
            var actual = this._sut.Say(number);

            // Assert
            Assert.AreEqual(expected, actual);
        }

        [Test]
        public void Say_Divisible5_ReturnBuzz()
        {
            // Arrange
            var number = 5;
            var expected = "Buzz";

            // Act
            var actual = this._sut.Say(number);

            // Assert
            Assert.AreEqual(expected, actual);
        }

        [Test]
        public void Say_Divisible3And5_ReturnFizzBuzz()
        {
            // Arrange
            var number = 15;
            var expected = "FizzBuzz";

            // Act
            var actual = this._sut.Say(number);

            // Assert
            Assert.AreEqual(expected, actual);
        }
    }
}
